src/main/java/com/tms/lesson52/hw/controller/StudentController.java

```diff
@@ -0,0 +1,147 @@
+    }
+
+    @GetMapping(value = "/getStudentByName")
+    public ModelAndView getStudentByName(@RequestParam(value = "name", required = false) String name, Model model) {
+        if (name != null) {
+            Student requestedStudent = studentRepository.findStudentByName(name);
+            model.addAttribute("student", requestedStudent);
+            return new ModelAndView("get/getStudent");
+        }
+        return new ModelAndView("get/getStudentByNameForm", "student", new Student());
+    }
+
+    @GetMapping(value = "/getStudentsByGroupId")
+    public ModelAndView getStudentsByGroupId(@RequestParam(value = "groupId", required = false) Integer groupId, ModelMap model) {
+        if (groupId != null) {
+            List<Student> requestedStudents = studentRepository.findStudentsByGroupId(groupId);
+            model.put("students", requestedStudents);
+            return new ModelAndView("get/getStudentsByGroupId");
+        }
+        return new ModelAndView("get/getStudentsByGroupIdForm", "students", new Student());
+    }
+
+    @GetMapping(value = "/getStudentsByPayment")
+    public ModelAndView getStudentsByPayment(@RequestParam(value = "ispaid", required = false) Boolean isPaid, ModelMap model) {
+        if (isPaid != null) {
+            List<Student> requestedStudents = studentRepository.findStudentsByPayment(isPaid);
+            System.out.println(requestedStudents);
+            model.put("students", requestedStudents);
+            return new ModelAndView("get/getStudentsByPayment");
+        }
+        return new ModelAndView("get/getStudentsByPaymentForm", "student", new Student());
+    }
+
+    @GetMapping(value = "/update")
+    public ModelAndView fillUpdateStudentForm() {
+        return new ModelAndView("update/updateStudentForm", "student", new Student());
+    }
+
+    @PostMapping(value = "/update")
+    public ModelAndView updateStudent(@ModelAttribute Student student, Model model) {
+        if (studentRepository.existsById(student.getId())) {
+            studentRepository.save(student);
+            String message = String.format("Student with '%s' id has been updated successfully!", student.getId());
+            model.addAttribute("message", message);
+            return new ModelAndView("update/updateStudent", "student", new Student());
+        }
+        return new ModelAndView("update/updateStudentForm", "student", new Student());
+    }
+
+    @GetMapping(value = "/updateStudentGroup")
+    public ModelAndView fillUpdateStudentGroupForm() {
+        return new ModelAndView("update/updateStudentGroupForm", "student", new Student());
+    }
+
+    @PostMapping(value = "/updateStudentGroup")
+    public ModelAndView updateStudentGroup(@ModelAttribute Student student, Model model) {
+        int id = student.getId();
+        int groupId = student.getGroupId();
+        if (studentRepository.existsById(id)) {
+            student = studentRepository.findById(id).orElse(null);
```

Comment:
Let's move it out of `if` and use null check instead of `studentRepository.existsById(id)` to avoid making second call to the database